STYLE.md

```diff
@@ -0,0 +1,364 @@
+ $(vopt_if colord 'colord-devel') $(vopt_if cups 'cups-devel')
+ $(vopt_if wayland 'libxkbcommon-devel wayland-devel wayland-protocols
+ libwayland-egl MesaLib-devel') $(vopt_if x11 'libXcursor-devel libXdamage-devel
+ libXext-devel libXinerama-devel libXi-devel libXrandr-devel
+ libXcomposite-devel')"
+depends="gtk-update-icon-cache shared-mime-info $(vopt_if x11 'dbus-x11')"
+```
+
+After the dependencies comes **short_desc**. Start it with an uppercase
+character, don't pass over 72 chars or use an article at the beginning.
+
+```
+# Wrong!
+# Superfluous usage of A
+short_desc="A configuration language guaranteed to terminate"
+# Correct!
+short_desc="Configuration language guaranteed to terminate"
+```
+
+**maintainer** follows. It uses a **name <email>** scheme, name and email must
+match the git equivalents used to commit. The email must be usable for contact
+and not try to be obscured.
+
+```
+# Wrong! (Email not reachable)
+maintainer="maxice8 <maxice8@github.noreply.com>"
+# Wrong! (Doesn't follow the scheme, missing <>)
+maintainer="maxice8 thinkabit.ukim@gmail.com"
+# Wrong! (Email obscured)
+maintainer="maxice8 <thinkabit.ukimATgmailDOTcom>"
+# Correct
+maintainer="maxice8 <thinkabit.ukim@gmail.com>"
+```
+
+Then there is **license** which should be specified according to the SPDX
+short-identifier. If multiple licenses apply then they should all be listed
+and each one be separated by a comma and a whitespace.
+
+```
+# Wrong!
+# GPL-2 isn't a SPDX short-identifier
+# there is no proper separation between the licenses
+license="GPL-2 MPL-2.0"
+# Correct!
+license="LGPL-2.1-or-later, Apache-2.0"
+```
+
+**homepage**, **changelog** and **distfiles** follow, if multiple distfiles are
+present then they should be broken with a newline, the same applies to
+**checksum**.
+
+```
+homepage="https://foosoft.com"
+changelog="https://foosoft.com/dist/foo/changelog.txt"
+distfiles="https://foosoft.com/dist/foo/${version}.tar.xz
+ https://foosoft.com/dist/bar/${version}.tar.xz"
+```
+
+Variables that are knobs to functionality and are not specific to a
+**build_style** should come at the end of the variables section.
+
+The same applies for variables that take structured values like **conf_files**.
+There is no specific ordering for such variables. but it is preferred that
+multi-line versions of **conf_files** and other variables that are prone to
+requiring multiple lines like **make_dirs** should be at the end, with a blank
+space separating them from the other variables.
+
+When using **nopie**, **nocross** and **broken** make the value be something
+descriptive, if no other description is available, a build log showcasing the
+failure is acceptable.
+
+```
+broken="Segfaults at runtime"
+nocross="Tries to execute binary built for target on host"
+```
+
+If **alternatives** is used it should be the last line, if there are multiple
+alternatives presented then they should be 1 on each line, prefixed with
+whitespace.
```

Comment:
Many, e.g. posix use word "whitespace" for any blank character, and "space" for most common of them. Is any whitespace allowed here?